### PR TITLE
Publish multi-arch images with goreleaser

### LINKED ISSUE #504: Switch to goreleaser to building and publishing images
**Describe the solution you'd like**
BotKube doesn't support multi-arch images as of now. With goreleaser it will ease the process of building multi-arch binaries/docker images and publishing it.

### PR DESCRIPTION
**Pull request by PrasadG193**
##### ISSUE TYPE
 - Feature Pull Request

##### SUMMARY
- Switch to goreleaser for building and publishing release
- Set BotKube version with build args
- Use docker manifest to build multi-arch image
- Update CI pipeline to build and push image using goreleaser

Fixes #504 

```
$ docker manifest inspect ghcr.io/prasadg193/botkube:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1370,
         "digest": "sha256:64d94709256eb85452afa06aa15574724266fa6b82b70654ff6b03c467d5a5e4",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:525d79e629637fd1ae3530223fa380aea26ed299a1fd9ae2787721f4cc33dad5",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:ef72bc09078929c8aed1161f86c2c86523f6fac664094a034c737ffd582f3ea7",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1160,
         "digest": "sha256:990cd2c6b522a3a69ba73ee83ed0ae306102369f02d09891de8fbfd024d37e87",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      }
   ]
}

```

### CODE REVIEW

**Comment by PrasadG193** [comment]
@xUnholy PTAL

**Comment by PrasadG193** [comment]
Ref: Building multi-arch images with goreleaser: https://carlosbecker.com/posts/multi-platform-docker-images-goreleaser-gh-actions/

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  use: buildx
+  goarch: arm
+  goarm: 7
+  dockerfile: 'build/Dockerfile'
+  build_flag_templates:
+  - "--platform=linux/arm"
+  - "--build-arg=botkube_version={{ .Tag }}"
+
+docker_manifests:
+- name_template: ghcr.io/infracloudio/botkube:{{ .Tag }}
```
Comment (sanketsudake) [inline .goreleaser.yml:54]: Final image which we are going to push would be following right ?
```docker tag infracloudio/botkube:latest-amd64 <your_account>/botkube:latest```

It wont have tags like `{{ .Tag }}-amd64` etc ?

File: .github/workflows/artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: goreleaser
```
Comment (sanketsudake) [inline .github/workflows/artifacts.yml:1]: can we rename it to `release-develop` or some proper name ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ Now you can build and run BotKube by one of the following ways
    ```sh
    $ make build
    $ make container-image
-   $ docker tag infracloudio/botkube:latest <your_account>/botkube:latest
+   $ docker tag infracloudio/botkube:latest-amd64 <your_account>/botkube:latest
```
Comment (sanketsudake) [inline CONTRIBUTING.md:36]: So for local building goreleaser would build all platform images ? can we restrict it to specific platforms ? also manage image tag internally.

File: .goreleaser.yml (hunk 1 above)
Comment (PrasadG193) [reply .goreleaser.yml:54]: @sanketsudake goreleaser will push all the images - `{{ .Tag }}-amd64`, `{{ .Tag }}-arm64` and `{{ .Tag }}-armv7`. And finally create a manifest with tag `{{ .Tag }}`. 
https://goreleaser.com/customization/docker_manifest/

File: CONTRIBUTING.md (hunk 3 above)
Comment (PrasadG193) [reply CONTRIBUTING.md:36]: Yes, goreleaser will build all the images. Image packaging does not take much time. If we don't want to use gorelease, we'll have to maintain a separate Dockerfile for a platform-specific build.

**Review by sanketsudake — APPROVED** [review]

File: .github/workflows/artifacts.yml (hunk 2 above)
Comment (PrasadG193) [inline .github/workflows/artifacts.yml:1]:
```suggestion
name: build-artifacts
```

**Pull request merged**